pkg/gofr/datasource/pubsub/mqtt/mqtt.go

```diff
@@ -338,19 +342,41 @@ func (m *MQTT) Unsubscribe(topic string) error {
 		close(sub.msgs)
 		delete(m.subscriptions, topic)
 	}
-	m.mu.RUnlock()
 
 	return nil
 }
 
-func (m *MQTT) Disconnect(waitTime uint) {
+func (m *MQTT) Close() error {
+	timeout := m.config.CloseTimeout
+
+	if timeout == 0 {
+		// setting default value if value is not populated
+		timeout = DefaultCloseTimeout
+	}
+
+	m.Client.Disconnect(uint(timeout.Milliseconds()))
```

Comment:
This should return the unsubscribe error.
```suggestion
	return m.Client.Disconnect(uint(timeout.Milliseconds()))
```

Reply:
sure